test/main_test.js

```diff
@@ -72,7 +72,7 @@ suite('main', function() {
       .then(() => fs.unlink(`${dir}/build.ninja`))
       .catch(() => { /* will fail if state is already clean */ })
       .then(() => {
-        return main({ dir: dir });
+        return main({ dir: dir, config: __dirname + '/fixtures/config.json' });
```

Comment:
Can you add another test case in `test/fixtures` that makes sure that all of the environment variables are set correctly when the configure script gets run? One way to do that is to write a build rule that throws an error or exits with a nonzero status code if some env var's value is wrong.